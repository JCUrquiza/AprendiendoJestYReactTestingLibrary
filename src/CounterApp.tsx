import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState( value );

    const aumentar = () => {
        setCounter(counter + 1);
    }

    const reducir = () => {
        setCounter(counter - 1);
    }

    const resetear = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>Counter App</h1>
            <h2>{ counter }</h2>

            <button onClick={ aumentar }>
                +1
            </button>
            <button onClick={ reducir }>
                -1
            </button>
            <button aria-label='btn-reset' onClick={ resetear }>
                Reset
            </button>

        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

