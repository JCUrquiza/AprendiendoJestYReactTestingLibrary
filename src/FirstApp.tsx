import PropTypes from 'prop-types';

/*
const getResult = () => {
    return 4 + 4;
}

interface Props {
    title: string;
    subTitle: string;
    name: string;
}
*/

export const FirstApp = ({ title, subTitle, name }) => {

    return (
        <>
            <h1>{ title }</h1>
            <p>{ subTitle }</p>
            <p>{ name }</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

FirstApp.defaultProps = {
    name: 'JC',
    subTitle: 'No hay Subtitulo'
    // title: 'No hay t√≠tulo',
}

