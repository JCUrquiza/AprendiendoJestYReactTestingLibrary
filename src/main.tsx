import React from 'react';
import ReactDOM from 'react-dom/client';
// import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloWorldApp';
import { CounterApp } from './CounterApp';
import './styles.css';

// { createRoot }


const appElement = document.getElementById('root');

if (appElement) {
  const appRoot = ReactDOM.createRoot(appElement);
  appRoot.render(
    <React.StrictMode>
      {/* 
      <HelloWorldApp />
      <FirstApp title='Hola soy JC' />
      */}
      <CounterApp value={ 100 } />
    </React.StrictMode>
  );
} else {
  console.error('No se encontr√≥ un elemento con id="app" en el DOM.');
}



/*
ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <HelloWorldApp />
        <FirstApp />
    </React.StrictMode>
)



const root = createRoot(document.getElementById('root'));
root.render(HelloWorldApp);
*/
