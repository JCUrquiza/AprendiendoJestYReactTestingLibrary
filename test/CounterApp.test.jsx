/** @jest-environment jsdom */
import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Prueba con <CounterApp />', () => {

    const initialValue = 10;

    test('Debe de hacer match con el snapshot', () => {

        // const { container } = render( <CounterApp value={ initialValue } /> );
        // expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar el valor inicial de 100 <CounterApp value={100}>', () => {

        // render( <CounterApp value={ 10 } /> );
        // expect( screen.getByText(10) ).toBeTruthy();
        
    });
    
    test('Debe de incrementar el botón +1', () => {
        
        // render( <CounterApp value={ initialValue } /> );
        // fireEvent.click( screen.getByText('+1') );
        // expect( screen.getByText('11') ).toBeTruthy();

    });

    test('Debe de decrementar el botón -1', () => {
        
        // render( <CounterApp value={ initialValue } /> );
        // fireEvent.click( screen.getByText('-1') );
        // expect( screen.getByText('9') ).toBeTruthy();
        
    });
    
    test('Debe de funcionar el botón de Reset', () => {
        
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );

        // fireEvent.click( screen.getByText('Reset') );
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
        // screen.debug();
        
        expect( screen.getByText( initialValue ) ).toBeTruthy();

    })

});



