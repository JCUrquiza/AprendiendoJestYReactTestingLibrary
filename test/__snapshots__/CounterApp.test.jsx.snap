// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con <CounterApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h1>
    0
  </h1>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
